core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -150,7 +154,8 @@ public <T extends RESTResponse> T execute(
     servletContext.setVirtualHosts(null);
     servletContext.setGzipHandler(new GzipHandler());
 
-    this.httpServer = new Server(8181);
+    initializePort();
+    this.httpServer = new Server(port);
```

Comment:
Jetty already provides functionality to bind to a random port by setting `this.httpServer = new Server(0);`. 

And then we can call `localPort()` in those 2 methods that require the port. The content of the method should be:
```
private int localPort() {
    assertThat(httpServer.isRunning()).isTrue();
    return ((ServerConnector) httpServer.getConnectors()[0]).getLocalPort();
  }
```

Therefore we don't need to introduce a `port` variable and also we don't need to find a random port ourselves